src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || c >= a + b) {
+      throw new Error(`Sides ${a}, ${b} and ${c} can't form a triangle`);
```

Comment:
The triangle inequality check should also include `a >= b + c` and `b >= a + c`. 📐

Reply:
Not resolved. Just replace condition with `a >= b + c` and `b >= a + c`